wandb/sdk/wandb_init.py

```diff
@@ -295,13 +298,34 @@ def _log_setup(self, settings):
 
         self._wl._early_logger_flush(logger)
 
-    def init(self):
+    def init(self):  # noqa: C901
         trigger.call("on_init", **self.kwargs)
         s = self.settings
         config = self.config
-
         if s._noop:
-            run = RunDummy()
+            # dict which allows . access to mimic Config
+            class Dict(dict):
+                def __getattr__(self, key):
+                    try:
+                        return object.__getattribute__(self, key)
+                    except AttributeError:
+                        return self[key]
+
+                def __setattr__(self, key, value):
+                    self[key] = value
+
+                def __getitem__(self, key):
+                    val = dict.__getitem__(self, key)
+                    if isinstance(val, dict):
+                        val = Dict(val)
+                        self[key] = val
+                    return val
```

Comment:
Let's put this Dict into the Dummy module.  We should definitely add a couple tests that assert the cases I mentioned.  I'm a little worried about all of this magic with `inspect` and setting globals.  But I suppose it's the most robust "noop" thing we could do.  @raubitsj do you have thoughts on the general approach?

Reply:
approach is good.

Reply:
I think these issues have been resolved.